simulator/func_sim/func_sim.cpp

```diff
@@ -56,23 +56,77 @@ typename FuncSim<ISA>::FuncInstr FuncSim<ISA>::step()
 }
 
 template <typename ISA>
-void FuncSim<ISA>::init( const std::string& tr)
+std::pair<Simulator::StopReason, TrapType> FuncSim<ISA>::run( uint64 instrs_to_run)
 {
+    if (!binary_file_loaded)
+        throw NoBinaryFile();
+
+    for ( uint64 i = 0; i < instrs_to_run; ++i) {
+        const auto& instr = step();
+        sout << instr << std::endl;
+
+        if ( instr.has_trap())
+            return std::make_pair( StopReason::TrapHit, instr.trap_type());
+        if ( instr.is_halt())
+            break; /* Can't return Halted here because of 'control reaches end of non-void' */
```

Comment:
More obvious to return something here and in the end of function instead of writing comments